bftengine/src/bftengine/ControllerWithSimpleHistory.cpp

```diff
@@ -121,6 +124,20 @@ bool ControllerWithSimpleHistory::onEndOfEvaluationPeriod() {
   bool downgraded = false;
   bool currentFirstPathChanged = false;
 
+  // Upgrade/Downgrade current path -
+  // If SLOW_PATH test for upgrade
+  //    Iterate over the last X(64) executions and count -
+  //    - How many times all replicas has participated in a transaction i.e. cyclesWithFullCooperation
+  //    - How many times more than 3*f + C and less than all, has participated i.e. cyclesWithPartialCooperation
+  //    If cyclesWithFullCooperation is greater than factorForFastOptimisticPath * EvaluationPeriod then upgrade to
+  //    OPTIMISTIC_FAST. Else - if C replicas is greater than 0 and cyclesWithFullCooperation +
```

Comment:
I think, instead of saying "if C replicas is greater than 0", you can simply say "C>0".